### PR TITLE
DEV into Master

### PR DESCRIPTION
**Pull request by RobertCzinege**
Hi,
Please evaluate this pull request from DEV branch to MASTER.

The most important changes:
- FEATURE: Client side routing with iron-pages
- FIX: Missing ajax components for local preview
- FIX: On /users the newly registered user was not on the list
- FIX: Some typos in the registration error message

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -6,9 +6,13 @@
 def index():
     return render_template('index.html')
 
-@app.route("/register")
-def register():
-    return render_template('registration.html')
+#@app.route("/users")
```
Comment (hubudibu) [inline server.py]: you can usually delete commented out code before a commit
if you feel you want to keep it for future reference, usually they were there in a previous commit, so git will remember anyways

File: static/custom-elements/email-input.html [hunk 2]
```diff
@@ -0,0 +1,216 @@
+<!--
+@license
+Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
```
Comment (hubudibu) [inline static/custom-elements/email-input.html]: if you create a new custom element based on / inspired by an official google element, consider updating / removing informative comments and copyright notices too

File: static/custom-elements/registration-form.html [hunk 3]
```diff
@@ -5,6 +5,10 @@
 <link rel="import" href="../bower_components/paper-icon-button/paper-icon-button.html">
 <link rel="import" href="../bower_components/paper-button/paper-button.html">
 <link rel="import" href="../bower_components/iron-ajax/iron-ajax.html">
+<link rel="import" href="../bower_components/gold-email-input/gold-email-input.html">
```
Comment (hubudibu) [inline static/custom-elements/registration-form.html]: are you still using gold-email-input?

**Comment by hubudibu** [comment]
- try doing smaller pull requests next time, e.g. one for each feature/fix
- merge master on this and resolve conflicts
- check line comments
then we're good!

**Comment by RobertCzinege** [comment]
I'd like to ask for review for pull request. 
Indicated conflicts are solved, line notes have been implemented where it was needed.
More comes with refactoring.

**Comment by hubudibu** [comment]
ok, LGTM

**Pull request merged**